test/test_unary_ufuncs.py

```diff
@@ -1,12 +1,151 @@
+import math
+from itertools import product, chain
+from numbers import Number
+
+import unittest
+
 import torch
 
 from torch.testing._internal.common_utils import \
-    (TestCase, run_tests)
+    (TestCase, run_tests, torch_to_numpy_dtype_dict, suppress_warnings,
+     IS_WINDOWS, TEST_NUMPY)
 from torch.testing._internal.common_methods_invocations import \
     (unary_ufuncs)
 from torch.testing._internal.common_device_type import \
-    (instantiate_device_type_tests, ops, onlyOnCPUAndCUDA, skipCUDAIfRocm)
+    (instantiate_device_type_tests, ops, onlyOnCPUAndCUDA, skipCUDAIfRocm,
+     dtypes)
+from torch.testing import \
+    (floating_types_and, integral_types, complex_types)
+
+if TEST_NUMPY:
+    import numpy as np
+
+# Tensor generators and helpers
+
+# Interesting values and extremal values for different dtypes
+_unsigned_int_vals = (0, 1, 55, 127)
+_int_vals = (0, -1, 1, -55, 55, -127, 127, -128, 128)
+_large_int_vals = (-1113, 1113, -10701, 10701)
+_float_vals = (0.,
+               -.001, .001,
+               -.25, .25,
+               -1., 1.,
+               -math.pi / 2, math.pi / 2,
+               -math.pi + .00001, math.pi - .00001,
+               -math.pi, math.pi,
+               -math.pi - .00001, math.pi + .00001)
+_large_float_vals = (-501, 501,
+                     -1001.2, 1001.2,
+                     -13437.7, 13437.7,
+                     -4988429.2, 4988429.2,
+                     -1e20, 1e20)
+_float_extremals = (float('inf'), float('-inf'), float('nan'))
+
+
+def _make_tensor(size, device, dtype, *, low, high):
+    if dtype is torch.bool:
+        return torch.randint(0, 2, size, device=device, dtype=dtype)
+
+    def _maybe_clamp(t, low_default, low, high_default, high):
+        low = low_default if low is None else max(low_default, low)
+        high = high_default if high is None else min(high_default, high)
+        if low != low_default or high != high_default:
+            return torch.clamp(t, low, high)
+        return t
+
+    if dtype is torch.uint8:
+        t = torch.randint(0, 10, size, device=device, dtype=dtype)
+        return _maybe_clamp(t, 0, low, 10, high)
+    elif dtype in integral_types():
+        t = torch.randint(-9, 10, size, device=device, dtype=dtype)
+        return _maybe_clamp(t, -9, low, 10, high)
```

Comment:
clamp is inclusive even for integers, so boundaries should be symmetric.

Reply:
Fixed.